mlx/backend/metal/resident.cpp

```diff
@@ -8,8 +11,20 @@ namespace mlx::core::metal {
 // TODO maybe worth including tvos / visionos
 #define supported __builtin_available(macOS 15, iOS 18, *)
 
+// Trying to create a residency set in a VM leads to errors.
+static bool in_vm() {
+  static int in_vm = -1;
+  if (in_vm != -1)
+    return in_vm;
+
+  std::size_t len = sizeof(in_vm);
+  sysctlbyname("kern.hv_vmm_present", &in_vm, &len, NULL, 0);
```

Comment:
Small nit: if you put this in a lambda then you don't have to do the check for `-1`. Something like:

```suggestion
  auto check_vm() {
    int inv_vm;
    std::size_t len = sizeof(in_vm);
    sysctlbyname("kern.hv_vmm_present", &in_vm, &len, NULL, 0);
    return in_vm
  };
  static int in_vm = check_vm();
  return in_vm;
```

Reply:
Thanks for the suggestion! I write basically no C++, so using a lambda didn't occur to me. Let me test this out locally and update this PR.